### DISCUSSION TITLE
customize sidebar scrollbar

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#208] [Ideas]
as I just realized when traversing the Dom in Inspector, we can have very nice themes and colors, but the sidebar's scroll handle isnt yet set:

<img width="580" alt="Scherm­afbeelding 2024-11-15 om 14 07 39" src="https://github.com/user-attachments/assets/8023b8ea-8a66-4633-85c0-0538367a3ac6">

it might be taking the color of another variable I didn't find yet, but it would be nice of we could set it to a certain color (variable)
as it belongs to the sidebar, wouldn't this be a nice setting for the plugin to control?

thoughts?

**Comment by elchininet** [comment]
>as I just realized when traversing the Dom in Inspector, we can have very nice themes and colors, but the sidebar's scroll handle isnt yet set:

Because the sidebar scrollbar is not coming from Home Assistant, it is a native scrollbar from your OS.

There is a CSS proposal that is not baseline yet, but is not supported by every browser (mainly Safari):

https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color

**Comment by Mariusthvdb** [comment]
I did find these pseudo classes in HA

<img width="261" alt="Scherm­afbeelding 2024-11-15 om 14 33 54" src="https://github.com/user-attachments/assets/819e7584-4b5e-441c-9144-2ce1d8920bcb">

and apparently we can set a background-color to the scrollbar-thumb:

<img width="256" alt="Scherm­afbeelding 2024-11-15 om 14 34 29" src="https://github.com/user-attachments/assets/795918b4-b9ff-4dcf-be96-d54e14791415">

must check what ha-styles.ts says about that......


apparently this: https://github.com/home-assistant/frontend/blob/991cf83ff3c14ca119c38aed4e1fc0f2af1e794e/src/resources/ha-style.ts#L37

```
  --scrollbar-thumb-color: rgb(194, 194, 194);
```

that looks promising.

check!

<img width="392" alt="Scherm­afbeelding 2024-11-15 om 14 39 16" src="https://github.com/user-attachments/assets/7a15fad8-c23f-4417-9d08-15fb32232c7c">


haha, it's as simple as setting that variable in the theme files

**Reply by elchininet** [reply]
Yes, that one is the same CSS [for webkit browsers only](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar). As the documentation states, it will work only for subset of users (for example, it will not work in Firefox).

**Reply by elchininet** [reply]
>haha, it's as simple as setting that variable in the theme files

If Home Assistant did the work of making it compatible with all the browsers it can be reused. It would be good to check how it behaves in other browsers / OS. If it is something that will work only for a few browsers / OS, it is good to keep it outside generic options (it would be a door for bug reports).

**Reply by elchininet** [reply]
I have checked and it seems that Home Assistant did the work of making it compatible with several browsers. I need to make a check among different browsers/OS but if it can be managed through CSS variables it is doable.

**Reply by Mariusthvdb** [reply]
yeah, well it's what HA uses, so we can at east play with that.

Ive just fired up firefox, and it nicely displays my

```
    scrollbar-thumb-color: var(--primary-color)
```

as does Safari (obviously ;-) ) Chrome, and lo and behold, even Microsoft Edge, all on Mac desktop.

for my testing purposes that will do.


btw, Ive updated my [Gist](https://gist.github.com/Mariusthvdb/e41957dc6f66d9764e0480030c9bd386) and added the remaining card-mod customizations below it in a comment.

could be helpful to people, feel free to reference it.

**Reply by elchininet** [reply]
Anything that has the class `ha-scrollbar` will get [these styles](https://github.com/home-assistant/frontend/blob/b8a13dd6ebdb50aad3fe820d9ace41152ec511ca/src/resources/styles.ts#L186-L203) and I am seeing that the `paper-listbox` element has it. I'll include an option for that variable but as it uses some `webkit` pseudo-classes, I'll add a small statement that this options is not guaranteed in every device.

**Reply by elchininet** [reply]
Test it with the last release:
https://github.com/elchininet/custom-sidebar/releases/tag/v6.7.0

**Reply by Mariusthvdb** [reply]
confirmed:

<img width="382" alt="Scherm­afbeelding 2024-11-16 om 09 34 55" src="https://github.com/user-attachments/assets/8a7c687f-dc96-4c9e-a5d5-e9414744e255">

it even overrides the setting for the same variable in my HA themes.

did you purposely choose the same name for that variable?
seems to me a dedicated `sidebar-xxx` name would be better in this case

**Reply by Mariusthvdb** [reply]
one things leads to another:
now we can color the thumb, the background of the scrollbar stands out being non themed.....
need to find the variable for that,  but my inspector wont see it somehow

**Reply by elchininet** [reply]
>did you purposely choose the same name for that variable?
>seems to me a dedicated sidebar-xxx name would be better in this case

Yes, I want to make it clear that it is not an implementation coming from the plugin. This could be a door for "the scroll is not colored in my device/browser" so I kept the same Home Assistant name and left a comment in the option.

**Reply by elchininet** [reply]
>one things leads to another:
now we can color the thumb, the background of the scrollbar stands out being non themed.....
>need to find the variable for that, but my inspector wont see it somehow

It is implemented, in the Home Assistant styles that I sent you they set [this property](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color) with the second value in transparent (that value is the one for the track), but of course it is not compatible with every browser. I think that they decided implementing the feature with [a graceful degradation](https://developer.mozilla.org/en-US/docs/Glossary/Graceful_degradation).
